src/scripts/autosplitter.gd

```diff
@@ -86,24 +77,19 @@ func _process(_delta: float) -> void:
 	if speedrun_is_running:
 		run_delta()
 		emit_signal("timer_updated")
-	#if server.is_listening():
-	#	server.poll()
-
-
-#func _on_server_client_close_request(id: int, code: int, reason: String) -> void:
-#	print("Received client close request: %d %d %s" % [id, code, reason])
-
-
-#func _on_server_client_connected(id: int, protocol: String) -> void:
-#	print("Client connected: %d %s" % [id, protocol])
-#	peers.append(id)
-#	server.get_peer(id).set_write_mode(WebSocketPeer.WRITE_MODE_TEXT)
-
-
-#func _on_server_client_disconnected(id: int, was_clean_close: bool) -> void:
-#	print("Client disconnected: %d %s" % [id, was_clean_close])
-#	peers.erase(id)
-
-
-#func _on_server_data_received(id: int) -> void:
-#	print("Data received: %d" % id)
+	if server.is_listening():
+		while server.is_connection_available():
+			var conn := server.take_connection()
+			var peer := WebSocketPeer.new()
+			var error := peer.accept_stream(conn)
+			if error:
+				push_warning("Incoming peer could not accept stream")
+			else:
+				print("Websocket connected: %s:%d" % [peer.get_connected_host(), peer.get_connected_port()])
+				peers.push_back(peer)
+		for i in range(peers.size() - 1, -1, -1): # Iterates through peers backwards, so they can be erased properly
```

Comment:
I've been a convert to the "always use a filter function when iterating through an array and deleting elements" school of thought, so something like

```
prune = func(peer: WebSocketPeer):
    peer.poll()
    if peer.get_ready_state() == WebSocketPeer.STATE_CLOSED:
        print("Disconnected.")
        return false
    return true
peers = peers.filter(prune)
```

easier to read IMO

Reply:
Lol good point, I actually had it similar to that but thought it was slightly weird for there to be a print statement inside the filter function. Seemed like there was a kind of "side effect" into the pure functional programming aspect of how filter works, but I suppose I'm just not experienced enough with using that in actual code to know if it's a normal thing to do. I'll add it back since I liked it better than my current approach.

Reply:
Yeah I think the line is if there's actual state information being changed as a result of the filter (besides changing the list, or garbage collection obv). Printing doesn't change state, so it's all good